rules/java_rules.build_defs

```diff
@@ -509,56 +513,65 @@ def _jarcat_cmd(main_class=None, preamble=None, manifest=None):
 
 # This function is exposed as java_toolchain unless Bazel compatibility is enabled in which case it is exposed as
 # please_java_toolchain. This is to avoid namespace conflicts with Bazel's java_toolchain rule. 
-def _java_toolchain(name : str, jdk_url : str|dict, visibility:list = ["PUBLIC"]) -> str:
+def _java_toolchain(name : str, jdk_url : str|dict, visibility:list = ["PUBLIC"], javac_worker:str=CONFIG.JAVAC_WORKER,
+                    download_hashes = []) -> str:
     """Downloads the JDK and sets up build rule metadata so the language rules can use the toolchain"""
-    current_package = package_name()
-
     jdk_url = jdk_url if isinstance(jdk_url, str) else jdk_url[CONFIG.OS]
 
     if not jdk_url:
         raise ParseError(f"No java jdk_url defined for platform {CONFIG.OS}")
 
+    this = canonicalise(f":{name}")
```

Comment:
do you need to canonicalise this? it shouldn't matter?

Reply:
Yeah we do, we're passing it via the metadata and it's used in the java_library and java_test rules.